test/test.js

```diff
@@ -385,7 +385,29 @@ describe('success tests', () => {
         .expect(validUserResponse)
         .end(done);
   });
-  
+
+  it('should work if secret is provided by secret callback', done => {
```

Comment:
We also need tests for two more scenarios:

1) Secret callback returns an error
2) Secret callback returns a valid secret but authentication fails